### DISCUSSION TITLE
WebEx Integration - Support in Latest version

### DISCUSSION THREAD

**Discussion by manjunathnayaka22** [grafana/grafana-infinity-datasource#535] [Q&A]
@yesoreyeram 
Could you please confirm if latest version of Infinity support WebEx API integration.  If not may I know how to send content-type method with application/x-www-form-urlencoded in Infinity? Thanks

**Comment by yesoreyeram** [comment]
> application/x-www-form-urlencoded

<img width="761" alt="image" src="https://user-images.githubusercontent.com/153843/220657737-1ba649dd-35d8-42d1-a8cf-435d39dff8ab.png">

**Reply by manjunathnayaka22** [reply]
@yesoreyeram Thank you for your quick response, with latest version of infinity I can see these option. I tried to send all necessary details to WebEx API but still getting an error- "error getting data frame. Unexpected end of JSON Input. 

I really appreciate if you could provide an option in fixing this issue within Infinity.

![image](https://user-images.githubusercontent.com/94129159/220890444-ecca2448-8e1a-41ae-8018-8ab02fd4834e.png)

**Reply by manjunathnayaka22** [reply]
@yesoreyeram If I choose type from JSON to HTML I dont see any error, however I am not getting actual response in the dashboard. 

![image](https://user-images.githubusercontent.com/94129159/220935710-2ce46ea7-db1f-4779-9f91-408bb619ec29.png)

Actual data which I am supposed to get from WebEx
![image](https://user-images.githubusercontent.com/94129159/220935840-92906991-5281-47f9-b596-f43635d7c5c8.png)

**Reply by yesoreyeram** [reply]
seems you are adding to headers instead form item.

**Comment by manjunathnayaka22** [comment]
@yesoreyeram apologies I am not getting which form item you are referring here. Could you please share more details on this.
Thanks

**Reply by yesoreyeram** [reply]
Parameters in form body not headers

**Comment by yesoreyeram** [comment]
I would like to reiterate.

There are few grant types in OAuth 2.0:

Authorization Code
Implicit
Resource Owner Password Credentials
Client Credentials

Webex currently only supports Authorization Code, which is a grant flow used mostly with web applications to get access to an API.

But **Authorization Code workflow require human interaction** to approve permissions which grafana doesnt support. Grafana only supports server to server authentication. 

Also the workaround you are trying is not secure as you are exposing the secure credentials in the the dashboard.
